evm/src/cpu/columns/mod.rs

```diff
@@ -159,14 +172,7 @@ pub struct CpuColumnsView<T: Copy> {
     pub(crate) general: CpuGeneralColumnsView<T>,
 
     pub(crate) clock: T,
-    /// 1 if this row includes a memory operation in the `i`th channel of the memory bus, otherwise
-    /// 0.
-    pub mem_channel_used: [T; memory::NUM_CHANNELS],
-    pub mem_is_read: [T; memory::NUM_CHANNELS],
-    pub mem_addr_context: [T; memory::NUM_CHANNELS],
-    pub mem_addr_segment: [T; memory::NUM_CHANNELS],
-    pub mem_addr_virtual: [T; memory::NUM_CHANNELS],
-    pub mem_value: [[T; memory::VALUE_LIMBS]; memory::NUM_CHANNELS],
+    pub mem_channels: [MemoryChannelView<T>; memory::NUM_CHANNELS],
```

Comment:
This is the _actual_ change; everything else is just fixing the resulting compiler errors.